layer/training/runtime/model_trainer.py

```diff
@@ -1,12 +1,19 @@
 import os
```

Comment:
Can we get the system metrics stuff outside of the `model_trainer` module, otherwise things get cluttered and more tightly coupled?

Reply:
You might be able to save me a lot of time, can you advise where you think this should be moved to?

Reply:
For example you could create a `layer/training/runtime/metrics.py` and define functions and classes there and import them here. You could create this metrics holding class that we talked about in another thread and that could be the coupling point between the modules, apart from starting the actual metrics tracking in a thread